packages/ic-http-certification/src/tree/certification_tree.rs

```diff
@@ -77,7 +77,10 @@ impl HttpCertificationTree {
 
                 // For wildcards we start by witnessing the path that was requested so the verifier can be sure
                 // that there is no exact match in the tree.
-                let requested_witness = self.tree.witness(&requested_tree_path);
+                let witness = self.tree.witness(&requested_tree_path);
+
+                // Then we witness the wildcard path that we will be responding with.
+                let witness = merge_hash_trees(witness, self.tree.witness(&responding_tree_path));
```

Comment:
The logic breaks on line 88. When the request path and the responding path have the same length, the loop never runs.
In the example the request path is `/` and the responding path is `/` (because we hit the wildcard).